File: 🟡 Medium/1590 - Make Sum Divisible by P/_1590MakeSumDivisiblebyP.py
class Solution:
    def minSubarray(self, nums: List[int], p: int) -> int:
        total_sum = sum(nums)
        sum_to_remove = total_sum % p

        if sum_to_remove == 0:
            return 0

        idx_map = {0: -1}
        curr_sum = 0
        res = len(nums)

        for i, num in enumerate(nums):
            curr_sum = (curr_sum + num) % p
            prefix = (curr_sum - sum_to_remove + p) % p

            if prefix in idx_map:
                res = min(res, i - idx_map[prefix])

            idx_map[curr_sum] = i

        if res == len(nums):
            return -1
        else:
            return res

File: 🟡 Medium/1590 - Make Sum Divisible by P/_1590MakeSumDivisiblebyP.js
/**
 * @param {number[]} nums
 * @param {number} p
 * @return {number}
 */
var minSubarray = function (nums, p) {
    let sum = 0;

    for (let num of nums) {
        sum = (sum + num) % p;
    }

    let sumToBeRemoved = sum % p;
    if (sumToBeRemoved == 0) {
        return 0;
    }

    const map = new Map();
    map.set(0, -1);

    let currSum = 0;
    let res = nums.length;
    for (let i = 0; i < nums.length; i++) {
        currSum = (currSum + nums[i]) % p;
        let prefix = (currSum - sumToBeRemoved + p) % p;

        if (map.has(prefix)) {
            res = Math.min(res, i - map.get(prefix));
        }

        map.set(currSum, i);
    }

    return res == nums.length ? -1 : res;
};